src/Nox.Types/Types/Temperature/Temperature.cs

```diff
@@ -1,9 +1,77 @@
+using Nox.Types.Common;
+using System.Collections.Generic;
+
 namespace Nox.Types;
 
+/// <summary>
+/// Represents a Nox <see cref="Temperature"/> type and value object.
+/// </summary>
+/// <remarks>Placeholder, needs to be implemented</remarks>
+public sealed class Temperature : Measurement<Temperature, TemperatureUnit>
+{
+
     /// <summary>
-    /// Represents a Nox <see cref="Temperature"/> type and value object.
+    /// Creates a new instance of <see cref="Temperature"/> object in <see cref="TemperatureUnit.Celsius"/>.
     /// </summary>
-    /// <remarks>Placeholder, needs to be implemented</remarks>
-    public sealed class Temperature : ValueObject<float, Temperature>
+    /// <param name="temperature">The value to create the <see cref="Temperature"/> with.</param>
+    /// <returns></returns>
+    /// <exception cref="TypeValidationException"></exception>
+    public static Temperature FromCelsius(QuantityValue temperature) => From(temperature, TemperatureUnit.Celsius);
+
+    /// <summary>
+    /// Creates a new instance of <see cref="Temperature"/> object in <see cref="TemperatureUnit.Fahrenheit"/>.
+    /// </summary>
+    /// <param name="temperature">The value to create the <see cref="Temperature"/> with.</param>
+    /// <returns></returns>
+    /// <exception cref="TypeValidationException"></exception>
+    public static Temperature FromFahrenheit(QuantityValue temperature) => From(temperature, TemperatureUnit.Fahrenheit);
+
+    /// <summary>
+    /// Creates a new instance of <see cref="Temperature"/> object in celsius.
+    /// </summary>
+    /// <param name="value">The value to create the <see cref="Temperature"/> with</param>
+    /// <returns></returns>
+    /// <exception cref="TypeValidationException"></exception>
+    public static new Temperature From(QuantityValue value)
+        => From(value, TemperatureUnit.Celsius);
+
+    /// <summary>
+    /// /// Validates a <see cref="Temperature"/> object.
+    /// </summary>
+    /// <returns>true if the <see cref="Temperature"/> value is valid.</returns>
+    internal override ValidationResult Validate()
     {
+        var result = Value.Validate();
+
+        if (double.IsNaN((double)Value) || double.IsInfinity((double)Value))
+        {
+            result.Errors.Add(new ValidationFailure(nameof(Value), $"Could not create a Nox Temperature type as value {Value} is not allowed."));
```

Comment:
Can we have a specific value NaN and Infinity, instead of {Value}, because the value will be evaluated differently based on culture and OS?

Reply:
In this case, you would have two checks, one for NaN and other for Infinity, similar to what we did in QuantityValue:
![image](https://github.com/NoxOrg/Nox.Generator/assets/30667828/43081d73-1982-42ee-9eb0-d064f91e2c75)